Failed login

Conor-Behard333 opened the pull request:
Test:
- pip install django-axes
- create a user
- login in with the username and wrong password 3 times
- should be locked out
- wait 5 minutes (can temporarily change to 1 minute for a faster test)
- login successfully
Details:
- added the django-axes module
- added functionality to lock the user out for 5 minutes if the user has 3 failed login attempt 
- updated readme

Conor-Behard333 committed:
User locking system

- added django-axes modules
- changes settings for axes in setting.py
- updated README

Conor-Behard333 committed:
Update README.md

Conor-Behard333 committed:
locks for 5 minutes

Conor-Behard333 committed:
changed to 3 failed attempts

Conor-Behard333 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Stop a user from logging in for 5 minutes after 3 failed attempts to log in (to protect against brute force attacks).](https://trello.com/c/NkGq0tXD/37-stop-a-user-from-logging-in-for-5-minutes-after-3-failed-attempts-to-log-in-to-protect-against-brute-force-attacks)

kfb19 reviewed (dismissed):
Code commented nicely but functionality is missing: 

Stop the need to refresh the page as this resets the timer. Some options are:
- Disable the button instead of redirecting to a new page 
- Have a counter telling them when to refresh so they don't guess and reset the timer 

> 

>

Conor-Behard333 committed:
returns to home page when failed login happens

Conor-Behard333 committed:
outputs message saying user has been blocked for 5 mins

- added message output in group_project1\base\templates\base\home.html

- added message when user is locked out in group_project1\base\views.py

- changed settings.py for axes

- added comments

Conor-Behard333 committed:
redirect to login page

added comment

Conor-Behard333 committed:
removed message output in home.html

Conor-Behard333 committed:
timer no longer resets after another failed login

- added login attempt model
- added functionality so that the user can see how long they are left out for

Conor-Behard333 committed:
updated display of lockout time to user

Conor-Behard333 committed:
added comments

Conor-Behard333 committed:
Merge branch 'main' into failed-login
